app/code/Magento/SalesRule/Model/CouponRepository.php

```diff
@@ -6,90 +6,63 @@
 
 /**
  * Coupon CRUD class
  *
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
  */
-class CouponRepository implements \Magento\SalesRule\Api\CouponRepositoryInterface
+class CouponRepository implements CouponRepositoryInterface
 {
-    /**
-     * @var \Magento\SalesRule\Model\CouponFactory
-     */
-    protected $couponFactory;
-
-    /**
-     * @var \Magento\SalesRule\Model\RuleFactory
-     */
-    protected $ruleFactory;
-
-    /**
-     * @var \Magento\SalesRule\Api\Data\CouponSearchResultInterfaceFactory
-     */
-    protected $searchResultFactory;
-
-    /**
-     * @var \Magento\SalesRule\Model\ResourceModel\Coupon\CollectionFactory
-     */
-    protected $collectionFactory;
-
-    /**
-     * @var \Magento\SalesRule\Model\Spi\CouponResourceInterface
-     */
-    protected $resourceModel;
-
-    /**
-     * @var \Magento\Framework\Api\ExtensionAttribute\JoinProcessorInterface
-     */
-    protected $extensionAttributesJoinProcessor;
-
-    /**
-     * @var \Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface
-     */
-    private $collectionProcessor;
-
     /**
      * CouponRepository constructor.
      * @param CouponFactory $couponFactory
      * @param RuleFactory $ruleFactory
-     * @param \Magento\SalesRule\Api\Data\CouponSearchResultInterfaceFactory $searchResultFactory
-     * @param ResourceModel\Coupon\CollectionFactory $collectionFactory
+     * @param CouponSearchResultInterfaceFactory $searchResultFactory
+     * @param CollectionFactory $collectionFactory
      * @param Spi\CouponResourceInterface $resourceModel
-     * @param \Magento\Framework\Api\ExtensionAttribute\JoinProcessorInterface $extensionAttributesJoinProcessor
+     * @param JoinProcessorInterface $extensionAttributesJoinProcessor
      * @param CollectionProcessorInterface|null $collectionProcessor
      */
     public function __construct(
-        \Magento\SalesRule\Model\CouponFactory $couponFactory,
-        \Magento\SalesRule\Model\RuleFactory $ruleFactory,
-        \Magento\SalesRule\Api\Data\CouponSearchResultInterfaceFactory $searchResultFactory,
-        \Magento\SalesRule\Model\ResourceModel\Coupon\CollectionFactory $collectionFactory,
-        \Magento\SalesRule\Model\Spi\CouponResourceInterface $resourceModel,
-        \Magento\Framework\Api\ExtensionAttribute\JoinProcessorInterface $extensionAttributesJoinProcessor,
-        CollectionProcessorInterface $collectionProcessor = null
+        protected readonly CouponFactory $couponFactory,
+        protected readonly RuleFactory $ruleFactory,
+        protected readonly CouponSearchResultInterfaceFactory $searchResultFactory,
+        protected readonly CollectionFactory $collectionFactory,
+        protected readonly CouponResourceInterface $resourceModel,
+        protected readonly JoinProcessorInterface $extensionAttributesJoinProcessor,
+        private ?CollectionProcessorInterface $collectionProcessor = null
```

Comment:
The type of this argument has changed by mistake.
```suggestion
        private CollectionProcessorInterface $collectionProcessor = null,
```